src/API.jl

```diff
@@ -67,12 +75,20 @@ function develop(ctx::Context, pkgs::Vector{PackageSpec}; shared::Bool=true,
 
     for pkg in pkgs
         # if julia is passed as a package the solver gets tricked
-        pkg.name != "julia" || pkgerror("Trying to develop julia as a package")
-        pkg.repo.rev === nothing || pkgerror("git revision can not be given to `develop`")
-        pkg.name !== nothing || pkg.uuid !== nothing || pkg.repo.url !== nothing ||
-            pkgerror("A package must be specified by `name`, `uuid`, `url`, or `path`.")
-        pkg.version == VersionSpec() ||
-            pkgerror("Can not specify version when tracking a repo.")
+        if pkg.name == "julia"
+            pkgerror("`julia` is not a valid package name.")
+        end
+        if pkg.name === nothing && pkg.uuid === nothing && pkg.repo.url === nothing
+            pkgerror("Packages must be specified by name, UUID, URL, or filesystem path.")
+        end
+        if pkg.repo.rev !== nothing
+            pkgerror("It is invalid to specify a git revision.",
```

Comment:
Not always so it would be good to be clear about why it is invalid here. Also, I personally prefer one sentence error messages, so perhaps something like

```
git revision cannot be specified in `develop` , got `$(pkg.repo.rev)` for package $(err_rep(pkg))
```

?

Reply:
Agree completely, I'll change it